test-error/apps/payslip-validator/.mvn/wrapper/MavenWrapperDownloader.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2007-present the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import java.net.*;
+import java.io.*;
+import java.nio.channels.*;
+import java.util.Properties;
+
+public class MavenWrapperDownloader {
```

Comment:
what exactly is the .mvn used for and why is it tracked by git? Is this some spring boot boilerplate? We already should have the mvn binary in the system path? Do we really need the mvnw wrapper and all these extra binaries and classes in the code and tracked by Git?

Reply:
Its a wrapper for the mvn binary.. that way we don't need to rely on the mvn version installed in the system. This way, each exercise is isolated from external dependencies, and it reduces possible problems with mvn versions, setups, etc. Is the way we do now to cover multiple OS and that way we control the maven version we want on the exercises. It's something we started doing on previous courses. We will discuss as team and take a decision.